docs/docker_host_ssh.md

```diff
@@ -0,0 +1,116 @@
+Reuse docker host SSH daemon
+============================
+
+It is possible to use the SSH daemon that runs on the docker host instead of using a separate SSH port for GitLab.
+
+## Setup
+
+Create the system `git` user if necessary (make sure it has a working shell):
+
+```bash
+useradd --system git --home-dir /srv/docker/gitlab/data
+```
+
+The home directory must point to GitLab data directory, as it takes `.ssh/authorized_keys` from there.
+
+Now create `gitlab-shell` home directory at the same location as in the container:
+
+```bash
+mkdir -p /home/git/gitlab-shell/bin
```

Comment:
I personally use incron permissions fixer on two different GitLab installs and it works fine. It's far from being elegant for sure, but it gets the job done.